From 107efb2f1a014c4475e122677faddbe0dbef1000 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Tue, 23 Jul 2024 14:54:59 +0200
Subject: [PATCH] add stack-protector option

Allow the user to disable stack-protector as it is not supported by all
toolchains

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Upstream: https://gitlab.gnome.org/GNOME/libdex/-/commit/a65d33aebbe43024a046b3f3cc4a1c6c533838d0
---
 meson.build       | 2 +-
 meson_options.txt | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index d0ca81d..2f7aac2 100644
--- a/meson.build
+++ b/meson.build
@@ -126,7 +126,7 @@ test_c_args = [
   ['-Werror=format-security', '-Werror=format=2'],
   '-FImsvc_recommended_pragmas.h',
 ]
-if get_option('buildtype') != 'plain'
+if get_option('buildtype') != 'plain' and get_option('stack-protector')
   if host_machine.system() != 'windows'
     test_c_args += '-fstack-protector-strong'
   endif
diff --git a/meson_options.txt b/meson_options.txt
index 519cd97..9e3364a 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -4,6 +4,9 @@ option('docs',
 option('examples',
        type: 'boolean', value: true,
        description: 'Build example programs')
+option('stack-protector',
+       type: 'boolean', value: true,
+       description: 'Enable stack-protector')
 option('vapi',
        type: 'boolean', value: true,
        description: 'Generate vapi data (requires vapigen)')
-- 
GitLab

